linera-execution/src/execution.rs

```diff
@@ -138,41 +139,49 @@ where
     ViewError: From<C::Error>,
     C::Extra: ExecutionRuntimeContext,
 {
+    #[allow(clippy::too_many_arguments)]
     async fn run_user_action(
         &mut self,
         application_id: UserApplicationId,
         chain_id: ChainId,
         action: UserAction,
         refund_grant_to: Option<Account>,
         grant: Option<&mut Amount>,
+        oracle_responses: Option<Vec<Vec<u8>>>,
         resource_controller: &mut ResourceController<Option<Owner>>,
-    ) -> Result<Vec<ExecutionOutcome>, ExecutionError> {
-        let execution_outcomes = match self.context().extra().execution_runtime_config() {
-            ExecutionRuntimeConfig::Synchronous => {
-                self.run_user_action_with_synchronous_runtime(
-                    application_id,
-                    chain_id,
-                    action,
-                    refund_grant_to,
-                    grant,
-                    resource_controller,
-                )
-                .await?
-            }
-        };
-        self.update_execution_outcomes_with_app_registrations(execution_outcomes)
-            .await
+    ) -> Result<(Vec<ExecutionOutcome>, Vec<Vec<u8>>), ExecutionError> {
+        let (execution_outcomes, oracle_responses) =
+            match self.context().extra().execution_runtime_config() {
+                ExecutionRuntimeConfig::Synchronous => {
+                    self.run_user_action_with_synchronous_runtime(
+                        application_id,
+                        chain_id,
+                        action,
+                        refund_grant_to,
+                        grant,
+                        oracle_responses,
+                        resource_controller,
+                    )
+                    .await?
+                }
+            };
+        let execution_outcomes = self
+            .update_execution_outcomes_with_app_registrations(execution_outcomes)
+            .await?;
+        Ok((execution_outcomes, oracle_responses))
     }
 
+    #[allow(clippy::too_many_arguments)]
     async fn run_user_action_with_synchronous_runtime(
         &mut self,
         application_id: UserApplicationId,
         chain_id: ChainId,
         action: UserAction,
         refund_grant_to: Option<Account>,
         grant: Option<&mut Amount>,
+        oracle_responses: Option<Vec<Vec<u8>>>,
         resource_controller: &mut ResourceController<Option<Owner>>,
-    ) -> Result<Vec<ExecutionOutcome>, ExecutionError> {
+    ) -> Result<(Vec<ExecutionOutcome>, Vec<Vec<u8>>), ExecutionError> {
```

Comment:
We may want to define a struct `Records` right away because there will be different types of records.

Reply:
Won't they always be `Vec<u8>`?
Happy to add a wrapper type, though.